### ISSUE TITLE
editor::SelectPrevious crashes in a multi buffer

### ISSUE THREAD

**Issue by 0x2CA** [zed-industries/zed#9766]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

As the title says

### Environment

Zed: v0.128.1 (Zed Preview)
OS: macOS 14.2.1
Memory: 32 GiB
Architecture: aarch64

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.

```
2024-03-25T10:57:11+08:00 [ERROR] {
  "thread": "main",
  "payload": "offset 18446744073702539879 is out of range, max allowed is 414933",
  "location_data": {
    "file": "crates/text/src/text.rs",
    "line": 2543
  },
  "backtrace": [
    "<backtrace::capture::Backtrace>::create",
    "<backtrace::capture::Backtrace>::new",
    "Zed::init_panic_hook::{closure#0}",
    "std::panicking::rust_panic_with_hook",
    "std::panicking::begin_panic_handler::{{closure}}",
    "std::sys_common::backtrace::__rust_end_short_backtrace",
    "_rust_begin_unwind",
    "core::panicking::panic_fmt",
    "<multi_buffer::Excerpt>::reversed_bytes_in_range",
    "<multi_buffer::MultiBufferSnapshot>::reversed_bytes_in_range::<usize>",
    "<editor::Editor>::select_previous",
    "<editor::Editor>::select_previous",
    "<gpui::elements::div::Interactivity>::on_action::<vim::visual::SelectPrevious, <gpui::window::ViewContext<workspace::Workspace>>::listener<vim::visual::SelectPrevious, <workspace::Workspace>::register_action<vim::visual::SelectPrevious, vim::visual::register::{closure#4}>::{closure#0}::{closure#0}>::{closure#0}>::{closure#0}",
    "<gpui::window::WindowContext>::dispatch_action_on_node",
    "<gpui::window::WindowContext>::dispatch_event",
    "<gpui::window::Window>::new::{closure#7}",
    "gpui::platform::mac::window::handle_key_event",
    "gpui::platform::mac::platform::send_event",
    "<gpui::platform::mac::platform::MacPlatform as gpui::platform::Platform>::run",
    "<gpui::app::App>::run::<Zed::main::{closure#3}>",
    "Zed::main",
    "std::sys_common::backtrace::__rust_begin_short_backtrace::<fn(), ()>",
    "std::rt::lang_start::<()>::{closure#0}",
    "std::rt::lang_start_internal",
    "_main"
  ],
  "app_version": "0.128.1",
  "release_channel": "Zed Preview",
  "os_name": "macOS",
  "os_version": "14.2.1",
  "architecture": "aarch64",
  "panicked_on": 1711335431868,
  "installation_id": "9590cfca-6a9b-4b08-aa60-0d9bc941ecd5",
  "session_id": "ed95aa47-46f8-44a8-b2c2-6df803c943ed"
}
```

**Comment by ConradIrwin** [comment]
Note, this is also reproducible with editor::SelectPrevious / ctrl-cmd-d

**Comment by JosephTLyons** [comment]
This is now fixed in [v0.130.0-pre](https://zed.dev/releases/preview/0.130.0).
